File: tests/fixtures/cpp/control_flow.cpp
// C++ control flow fixture: if, for, while, and range-based for loops.
#include <string>

int classify(int x) {
    if (x > 0) {
        return 1;
    } else if (x < 0) {
        return -1;
    } else {
        return 0;
    }
}

int sumTo(int n) {
    int total = 0;
    for (int i = 0; i < n; i++) {
        total = total + i;
    }
    return total;
}

void countdown(int start) {
    while (start > 0) {
        start = start - 1;
    }
}

int sumArray(int arr[], int len) {
    int sum = 0;
    for (int i = 0; i < len; i++) {
        sum = sum + arr[i];
    }
    return sum;
}

std::string joinWords(std::string words[], int count) {
    std::string result = "";
    for (int i = 0; i < count; i++) {
        result = result + words[i];
    }
    return result;
}

File: tests/fixtures/rust/control_flow.rs
fn classify(n: i32) -> i32 {
    if n > 0 {
        return 1;
    } else if n < 0 {
        return -1;
    } else {
        return 0;
    }
}

fn sum_range(limit: i32) -> i32 {
    let mut total = 0;
    for i in 0..limit {
        total = total + i;
    }
    return total;
}

fn count_down(start: i32) -> i32 {
    let mut n = start;
    while n > 0 {
        n = n - 1;
    }
    return n;
}

fn first_even(items: i32) -> i32 {
    let mut i = 0;
    loop {
        if i % 2 == 0 {
            return i;
        }
        i = i + 1;
    }
}

fn label(n: i32) -> i32 {
    match n {
        1 => 10,
        2 => 20,
        _ => 0,
    }
}
